File: Weknot/app/src/main/java/com/example/weknot/api/UserApi.java
package com.example.weknot.api;

import com.example.weknot.data.User;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface UserApi {

    @FormUrlEncoded
    @POST("")
    Call<User> getUser(@Field("id") String id);

}

File: weknot_android/app/src/main/java/com/example/weknot_android/network/api/UserApi.kt
package com.example.weknot_android.network.api

import com.example.weknot_android.model.user.Profile
import com.example.weknot_android.network.response.Response
import io.reactivex.Single
import retrofit2.http.*

interface UserApi {

    @GET("users/profile/{userId}")
    fun getProfile(@Header("Authorization") token: String, @Path("userId") userId: String): Single<retrofit2.Response<Response<Profile>>>
}